File: app/src/main/java/com/wadektech/eventshub/adapter/SocialEventsAdapter.java
package com.wadektech.eventshub.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.wadektech.eventshub.R;
import com.wadektech.eventshub.models.SocialEvents;
import java.util.List;

public class SocialEventsAdapter extends RecyclerView.Adapter<SocialEventsAdapter.ViewHolder> {
    public List<SocialEvents> socialEventsList ;
    public Context context ;
    public OnSingleSocialEventClicked singleSocialEventClicked ;

    public SocialEventsAdapter(List<SocialEvents> socialEventsList, Context context, OnSingleSocialEventClicked singleSocialEventClicked) {
        this.socialEventsList = socialEventsList;
        this.context = context;
        this.singleSocialEventClicked = singleSocialEventClicked;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.events_list, parent, false);
        return new ViewHolder(view , singleSocialEventClicked);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        SocialEvents socialEvents = socialEventsList.get(position);

        holder.date.setText(String.valueOf(socialEvents.getDate()));
        holder.entryFees.setText(String.valueOf(socialEvents.getTitle()));
        holder.eventTitle.setText(socialEvents.getTitle());
        holder.location.setText(socialEvents.getLocation());
        holder.shortDescription.setText(socialEvents.getShortDesc());
    }

    @Override
    public int getItemCount() {
        return socialEventsList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView eventTitle , shortDescription, date, location , entryFees ;
        OnSingleSocialEventClicked socialEventClicked ;
        public ViewHolder(@NonNull View itemView , OnSingleSocialEventClicked socialEventClicked) {
            super(itemView);
            eventTitle = itemView.findViewById(R.id.tv_event_title);
            shortDescription = itemView.findViewById(R.id.tv_title_desc);
            date = itemView.findViewById(R.id.tv_event_date);
            location = itemView.findViewById(R.id.tv_event_location);
            entryFees = itemView.findViewById(R.id.tv_event_fees);
            this.socialEventClicked = socialEventClicked ;
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            socialEventClicked.onSingleSocialEventItemClicked(getAdapterPosition());
        }
    }

    public interface OnSingleSocialEventClicked{
        void onSingleSocialEventItemClicked(int position) ;
    }
}

File: app/src/main/java/com/wadektech/eventshub/adapter/SocialEventsAdapter.kt
package com.wadektech.eventshub.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.paging.PagedListAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.wadektech.eventshub.R
import com.wadektech.eventshub.databinding.SocialEventsListBinding
import com.wadektech.eventshub.models.SocialEvents

class SocialEventsAdapter(var singleSocialEventClicked: OnSingleSocialEventClicked) :
        PagedListAdapter<SocialEvents, SocialEventsAdapter.ViewHolder>(SocialDiffUtil()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder.from(parent)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val socialEvents = getItem(position)
        if (socialEvents!=null){
            holder.bind(socialEvents, singleSocialEventClicked)
        }
    }

    class ViewHolder private constructor(val binding: SocialEventsListBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(socialEvents: SocialEvents, socialEventClicked: OnSingleSocialEventClicked) {
            binding.tvEventTitle.text = socialEvents.title
            binding.tvTitleDesc.text = socialEvents.shortDesc
            binding.tvEventDate.text = socialEvents.date.toString()
            binding.tvEventLocation.text = socialEvents.location
            binding.tvEventFees.text = socialEvents.entryFees.toString()

            itemView.setOnClickListener {
                socialEventClicked.onSingleSocialEventItemClicked(adapterPosition)
            }
        }
        companion object {
            fun from(parent: ViewGroup): ViewHolder {
                val layoutInflater = LayoutInflater.from(parent.context)
                val binding = SocialEventsListBinding.inflate(layoutInflater , parent, false)
                return ViewHolder(binding)
            }
        }
    }
        interface OnSingleSocialEventClicked {
            fun onSingleSocialEventItemClicked(position: Int)
        }

    class SocialDiffUtil : DiffUtil.ItemCallback<SocialEvents>(){
        override fun areItemsTheSame(oldItem: SocialEvents, newItem: SocialEvents): Boolean {
            return newItem.id == oldItem.id
        }

        override fun areContentsTheSame(oldItem: SocialEvents, newItem: SocialEvents): Boolean {
            return newItem.id == oldItem.id
        }
    }
}